src/root.zig

```diff
@@ -127,11 +127,13 @@ pub fn AlgorithmType(
 
         pub fn init(
             allocator: std.mem.Allocator,
+            max_haystack: usize,
+            max_needle: usize,
         ) !Self {
 
             // init to a min size, and resize after if needed
-            const rows = 1;
-            const cols = 1;
+            const rows = max_needle + 1;
+            const cols = max_haystack + 1;
 
             var m = try Matrix.init(allocator, rows, cols);
```

Comment:
```suggestion
            var m = try Matrix.init(allocator, max_haystack, max_needle);
```

Reply:
Oh I thought you were calling the `resize` function for some reason, where you already have the `+1`. I see now that this is the original init function. In that case, yes you are correct, ignore this suggestion.